csrc/id_model/predicate_indexing.cpp

```diff
@@ -225,14 +225,21 @@ std::unordered_map<Val*, Val*> getPredicateIndexReplacementMap(
   // main loop of circular buffering, increment the index by
   // (number_of_stages - 1) since the main loop has a read that is
   // (number_of_stages - 1) elements ahead.
+  // Only required for pipelined circular buffering, for warp specialized
+  // circular buffering, there is no prologue or epilog loop.
   auto replace_for_circular_buffering = [&](ForLoop* fl,
                                             Val* original_index) -> Val* {
     auto circular_buffer_axis =
         GpuLower::current()->circularBufferInfo().getCircularBufferAxis(tv);
     if (circular_buffer_axis == nullptr ||
         !id_model.idGraph(IdMappingMode::LOOP)
              .disjointValSets()
-             .strictAreMapped(fl->iter_domain(), circular_buffer_axis)) {
+             .strictAreMapped(fl->iter_domain(), circular_buffer_axis) ||
+        std::holds_alternative<WarpSpecialized>(
```

Comment:
We have a helper function for this in circular buffering pass. It would be cleaner to move to `lower_utils` and use it here.

https://github.com/NVIDIA/Fuser/blob/main/csrc/device_lower/pass/circular_buffer.cpp#L1244-L1250